libkbfs/key_bundle_cache.go

```diff
@@ -24,21 +25,27 @@ type keyBundleCacheKey struct {
 	isWriter    bool
 }
 
-// NewKeyBundleCacheStandard constructs a new KeyBundleCacheStandard with the given
-// cache capacity.
-func NewKeyBundleCacheStandard(capacity int) *KeyBundleCacheStandard {
-	head, err := lru.New(capacity)
-	if err != nil {
-		panic(err.Error())
-	}
-	return &KeyBundleCacheStandard{head}
+func (k keyBundleCacheKey) str() string {
+	return k.tlf.String() + k.bundleIDStr + strconv.FormatBool(k.isWriter)
+}
+
+// NewKeyBundleCacheLRU constructs a new KeyBundleCacheStandard with LRU
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheLRU(capacityBytes int) *KeyBundleCacheStandard {
+	return &KeyBundleCacheStandard{cache.NewLRUEvictedCache(capacityBytes)}
+}
+
+// NewKeyBundleCacheRandom constructs a new KeyBundleCacheStandard with random
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheRandom(capacityBytes int) *KeyBundleCacheStandard {
+	return &KeyBundleCacheStandard{cache.NewRandomEvictedCache(capacityBytes)}
 }
 
 // GetTLFReaderKeyBundle implements the KeyBundleCache interface for KeyBundleCacheStandard.
 func (k *KeyBundleCacheStandard) GetTLFReaderKeyBundle(
 	tlf tlf.ID, bundleID TLFReaderKeyBundleID) (*TLFReaderKeyBundleV3, error) {
-	cacheKey := keyBundleCacheKey{tlf, bundleID.String(), false}
-	if entry, ok := k.lru.Get(cacheKey); ok {
+	cacheKey := keyBundleCacheKey{tlf, bundleID.String(), false}.str()
```

Comment:
Oh sorry just realized I missed this comment. I guess I felt `string` was slightly lighter than a `Measurable` interface. Also an interface value means it can potentially be not comparable, making then not suitable for key type for `map`, and would cause a panic if such keys are passed in. We could check if they are comparable but that would require reflect.